gtsfm/frontend/global_descriptor/netvlad_global_descriptor.py

```diff
@@ -36,10 +38,15 @@ def describe(self, image: Image) -> np.ndarray:
         Returns:
             img_desc: array of shape (D,) representing global image descriptor.
         """
-        # initializing in the constructor leads to OOM.
-        model: nn.Module = NetVLAD()
-
-        img_tensor = torch.from_numpy(image.value_array).permute(2, 0, 1).unsqueeze(0).type(torch.float32) / 255
+        # Load model.
+        # Note: Initializing in the constructor leads to OOM.
+        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        model: nn.Module = NetVLAD().to(device)
+        model.eval()
+
+        img_tensor = (
+            torch.from_numpy(image.value_array).to(device).permute(2, 0, 1).unsqueeze(0).type(torch.float32) / 255
+        )
```

Comment:
could we wrap the model forward() call in `torch.no_grad()`?

Reply:
Done